b2sdk/bucket.py

```diff
@@ -354,6 +359,32 @@ def ls(
         if prefix != '' and not prefix.endswith('/'):
             prefix += '/'
 
+        # If we're running with wildcard-matching, we could get
+        # a different prefix from it.  We search for the first
+        # occurrence of any of the special characters and fetch
+        # parent path from that place.
+        # Examples:
+        #   'b/c/*.txt' –> 'b/c/'
+        #   '*.txt' –> ''
+        #   'a/*/result.[ct]sv' –> 'a/'
+        if with_wildcard:
+            for wildcard_character in ['*', '?', '[']:
```

Comment:
```suggestion
            for wildcard_character in '*?[':
```
lets not allocate 2KB for a string with len() of 3 :)